driver/web/rest/adminapis.go

```diff
@@ -1387,3 +1387,460 @@ func (h AdminApisHandler) GetContentItemsCategories(claims *tokenauth.Claims, w
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json; charset=utf-8")
+	w.WriteHeader(http.StatusOK)
+	w.Write(jsonData)
+}
+
+// GetCategory Gets a category
+// @Description Gets a category
+// @Tags Admin
+// @ID AdminGetCategory
+// @Accept json
+// @Produce json
+// @Success 200
+// @Security AdminUserAuth
+// @Router /admin/category/{id} [get]
+func (h AdminApisHandler) GetCategory(claims *tokenauth.Claims, w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	id := vars["id"]
+
+	resData, err := h.app.Services.GetCategory(claims.AppID, claims.OrgID, id)
+	if err != nil {
+		log.Printf("Error on getting category with name - %s\n %s", id, err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	data, err := json.Marshal(resData)
+	if err != nil {
+		log.Println("Error on marshal of category")
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json; charset=utf-8")
+	w.WriteHeader(http.StatusOK)
+	w.Write(data)
+}
+
+// UpdateCategory Updates a category with a specificed id.
+// @Description  Updates a category with a specificed id.
+// @Tags Admin
+// @ID AdminUpdateCategory
+// @Accept json
+// @Produce json
+// @Success 200 {object} model.Category
+// @Security AdminUserAuth
+// @Router /admin/category [put]
+func (h AdminApisHandler) UpdateCategory(claims *tokenauth.Claims, w http.ResponseWriter, r *http.Request) {
+
+	data, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		log.Printf("Error on marshal create a data content item - %s\n", err.Error())
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		return
+	}
+
+	var body model.Category
+	err = json.Unmarshal(data, &body)
+	if err != nil {
+		log.Printf("Error on unmarshal the update category request data - %s\n", err.Error())
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	resData, err := h.app.Services.UpdateCategory(claims.AppID, claims.OrgID, &body)
+	if err != nil {
+		log.Printf("Error on updating category  - %s", err)
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	jsonData, err := json.Marshal(resData)
+	if err != nil {
+		log.Println("Error on marshal the updated category")
+		http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+		return
+	}
+
+	w.Header().Set("Content-Type", "application/json; charset=utf-8")
+	w.WriteHeader(http.StatusOK)
+	w.Write(jsonData)
+}
+
+// DeleteCategory Deletes a category with specified id
+// @Description Deletes a category with specified id
+// @Tags Admin
+// @ID AdminDeleteCategory
+// @Success 200
+// @Security AdminUserAuth
+// @Router /admin/category/{id} [delete]
+func (h AdminApisHandler) DeleteCategory(claims *tokenauth.Claims, w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	id := vars["id"]
```

Comment:
Let's make this path parameter `name` because that is what we are searching by in Mongo.